src/Assert.php

```diff
@@ -873,6 +922,8 @@ public static function uniqueValues(array $values, $message = '')
     }
 
     /**
+     * @psalm-pure
```

Comment:
In the same manner as the filter_var based assertions can't be pure. object comparison with  `!=` can also be impure: https://3v4l.org/saUIr

Reply:
Oh shit, I didn't know string casts were performed on `==` and `!=`! Crap, getting rid of it :S

Reply:
a126a05afbc1433428d39676ce68e92fc9040a0c